File: wiki_examples/src/main/java/com/github/klee0kai/wiki/wrapping/TechFactoryComponent.java
package com.github.klee0kai.wiki.wrapping;

import com.github.klee0kai.stone.annotations.component.Component;
import com.github.klee0kai.stone.wrappers.LazyProvide;
import com.github.klee0kai.stone.wrappers.PhantomProvide;
import com.github.klee0kai.test.tech.components.Battery;
import com.github.klee0kai.test.tech.components.Ram;

@Component(wrapperProviders = {CustomWrapper.class})
public interface TechFactoryComponent {

    TechFactoryModule factory();

    PhantomProvide<Battery> battery();

    LazyProvide<Ram> ramMemory();

    void inject(GoodPhone goodPhone);
}

File: wiki_kotlin/src/main/kotlin/com/github/klee0kai/wiki/wrapping/TechFactoryComponent.kt
package com.github.klee0kai.wiki.wrapping

import com.github.klee0kai.stone.Provider
import com.github.klee0kai.stone.annotations.component.Component
import com.github.klee0kai.stone.wrappers.LazyProvider
import com.github.klee0kai.test.tech.components.Battery
import com.github.klee0kai.test.tech.components.Ram

@Component(
    wrapperHelpers = [CustomLazyWrapper::class],
)
interface TechFactoryComponent {

    fun factory(): TechFactoryModule

    fun battery(): Provider<Battery>

    fun ramMemory(): LazyProvider<Ram>

    fun inject(goodPhone: GoodPhone)

}
